mlflow/utils/annotations.py

```diff
@@ -28,6 +29,7 @@ def deprecated_func(func):
         if alternative is not None and alternative.strip():
             notice += " Use ``%s`` instead." % alternative
         func.__doc__ = notice + "\n" + func.__doc__
+        warnings.warn(notice, stacklevel=2)
```

Comment:
I don't think this does exactly what we intend. As soon as the module containing the decorated function is imported, this warning is executed regardless of whether or not the user invoked the decorated function.

Reply:
Good call - I updated the decorator to warn only on function usage (not import) and to warn each unique warning only once. Turns out it doesn't do this by default.